### ISSUE TITLE
Add documentation for TCP6ServerEndpoint

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5741]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)| @ashfall reported|
|-|-|
|Trac ID|trac#5741|
|Type|task|
|Created|2012-07-08 14:02:13Z|
|Branch|https://github.com/twisted/twisted/tree/add-howto-tcp6-endpoint-5741|

The [howto](http://twistedmatrix.com/documents/current/core/howto/endpoints.html) on endpoints needs to be updated to include the recently merged TCP6ServerEndpoint.

<details><summary>Searchable metadata</summary>

```
trac-id__5741 5741
type__task task
reporter__ashfall ashfall
priority__normal normal
milestone__ 
branch__branches_add_howto_tcp6_endpoint_5741 branches/add-howto-tcp6-endpoint-5741
branch_author__ashfall ashfall
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1341756133000000 1341756133000000
changetime__1345476339000000 1345476339000000
version__None None
owner__ashfall ashfall
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_1"></a>|@ashfall commented|
|-|-|

(In [34718]) Branching to 'add-howto-tcp6-endpoint-5741'

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_2"></a>|@ashfall removed owner|
|-|-|

Though TCP6ServerEndpoint is not majorly different from the TCP IPv4 endpoint, I hope the format for this addition is fine.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @ashfall|
|-|-|

The example string doesn't work. :)  ':' need to be escaped in string endpoint descriptions.  Also, whose IP address is that?  Twisted documentation should use unassigned numbers/names, or numbers/names explicitly assigned for example or local usage.  An fe80 address would probably be suitable.  If that 2001 address is such an address, feel free to correct me.  It is perhaps now also worth pointing out that the TCP server endpoint is IPv4-only.

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_4"></a>|@ashfall removed owner|
|-|-|

Replying to [exarkun](#note_3):

Thanks for the review. Changes made.

> Also, whose IP address is that?  Twisted documentation should use unassigned numbers/names, or numbers/names explicitly assigned for example or local usage.  An fe80 address would probably be suitable.  If that 2001 address is such an address, feel free to correct me.

[RFC 3849](http://tools.ietf.org/html/rfc3849) says the prefix 2001:DB8::/32 is allocated for documentation purposes. If you'd still prefer an fe80 address, let me know.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @ashfall|
|-|-|

Thanks for working on this, ashfall. I concur with your reading of the RFC on reserved prefixes. Two tiny things:
 * Can you linewrap the TCP (IPv4) `li` to < 80 characters like the rest of the document?
 *  While you're working on this document, can you fix the broken link to http://twistedmatrix.com/documents/12.1.0/api/twisted.internet.interfaces.IClientStreamEndpoint.connect.html on line 117.

Other than that, looks good, please merge!

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_6"></a>|@ashfall set status to `closed`|
|-|-|

(In [35309]) Merge add-howto-tcp6-endpoint-5741: Added documentation for TCP6ServerEndpoint

Author: ashfall
Reviewer: exarkun, jesstess
Fixes: [#5741](https://github.com/twisted/twisted/issues/5741)

Updated the howto on endpoints to include details of TCP6ServerEndpoint.
Also, the howto now points out that the TCP server endpoint is IPv4-only.
